services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt

```diff
@@ -72,563 +72,597 @@ class HankeControllerITests(@Autowired override val mockMvc: MockMvc) : Controll
+                hankeService.createHanke(any())
+                permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET)
+                disclosureLogService.saveDisclosureLogsForHanke(createdHanke, USERNAME)
+            }
+        }
 
-        verify { hankeService.createHanke(any()) }
-        verify { disclosureLogService.saveDisclosureLogsForHanke(any(), any()) }
-        verify { permissionService.setPermission(any(), any(), any()) }
-    }
+        @Test
+        fun `When perustaja is provided return provided information`() {
+            val hanke = HankeFactory.create().withPerustaja()
+            every { hankeService.createHanke(any()) } returns hanke
+            justRun { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
+
+            post(url, hanke)
+                .andExpect(status().isOk)
+                .andExpect(jsonPath("$.perustaja.nimi").value("Pertti Perustaja"))
+                .andExpect(jsonPath("$.perustaja.email").value("foo@bar.com"))
+
+            verifySequence {
+                hankeService.createHanke(any())
+                permissionService.setPermission(any(), any(), any())
+                disclosureLogService.saveDisclosureLogsForHanke(any(), any())
+            }
+        }
 
-    @Test
-    fun `createHanke sanitizes hanke input returns 200`() {
-        val hanke = HankeFactory.create().apply { generated = true }
-        every { hankeService.createHanke(hanke.copy(id = null, generated = false)) } returns
-            hanke.copy(generated = false)
-        justRun { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
+        @Test
+        fun `Sanitize hanke input and return 200`() {
+            val hanke = HankeFactory.create().apply { generated = true }
+            every { hankeService.createHanke(hanke.copy(id = null, generated = false)) } returns
+                hanke.copy(generated = false)
+            justRun { permissionService.setPermission(any(), any(), Role.KAIKKI_OIKEUDET) }
 
-        post(BASE_URL, hanke).andExpect(status().isOk)
+            post(url, hanke).andExpect(status().isOk)
 
-        verify { hankeService.createHanke(any()) }
-        verify { disclosureLogService.saveDisclosureLogsForHanke(any(), any()) }
-        verify { permissionService.setPermission(any(), any(), any()) }
-    }
+            verifySequence {
+                hankeService.createHanke(any())
+                permissionService.setPermission(any(), any(), any())
+                disclosureLogService.saveDisclosureLogsForHanke(any(), any())
+            }
+        }
 
-    @Test
-    fun `create hanke with perustaja without sahkoposti returns 400`() {
-        val hakemus = HankeFactory.create().withPerustaja()
-        val content: ObjectNode = OBJECT_MAPPER.valueToTree(hakemus)
-        (content.get("perustaja") as ObjectNode).remove("sahkoposti")
+        @Test
+        @Disabled("Seems like this is not implemented yet.")
```

Comment:
At a quick glance, this may not be the only missing validation. If that is the case, we should create a ticket of the required fix. Otherwise it is likely that this test will remain disabled for quite some time.

Reply:
There are tickets for perustaja:
- [HAI-1348](https://helsinkisolutionoffice.atlassian.net/browse/HAI-1348) for saving the perustaja info in FE.
- [HAI-1350](https://helsinkisolutionoffice.atlassian.net/browse/HAI-1350) for validating that the info is set in BE.

I don't what know other missing validations you're referring to.

Reply:
Ah yes, I see now. The reason for the error is that the email field is non-nullable. So we don't need validation to catch the missing value.

The difference between HAI0003 and HAI1002 is that for HAI0003 we can't even read the request to start checking its validity. It's incorrect here; the missing email should be a validation error. But I think that would require a separate DTO with nullable fields or a custom deserializer.

I think this PR has strayed too far from the original issue already, so fixing the validation won't be done here. And as long as nothing really uses the codes, it's not critical to have the correct code in every situation, as long as the HTTP response code is correct.

I fixed this to test the current implementation and re-enabled the test.